### PR TITLE
feat: inherit rc-pagination props, more customizable

### PR DESCRIPTION
**Pull request by JounQin**
close #32128, #32129

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

#32128, #32129

### 💡 Background and solution

As tile

### 📝 Changelog

| Language   | Changelog                                            |
| ---------- | ---------------------------------------------------- |
| 🇺🇸 English | feat: inherit rc-pagination props, more customizable |
| 🇨🇳 Chinese | 特性：继承 rc-pagination 的 props，更加可定制化      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by JounQin** [comment]
@afc163 

Although I don't know why `LocaleReceiver`'s types are affected, but I just fixed the previous issue.

**Comment by JounQin** [comment]
> @afc163
> 
> Although I don't know why `LocaleReceiver`'s types are affected, but I just fixed the previous typing issue.

ping @afc163 The CI is broken due to `LocaleReceiver`.

File: components/table/Table.tsx [hunk 1]
```diff
@@ -218,7 +218,7 @@ function Table<RecordType extends object = any>(props: TableProps<RecordType>) {
 
       // Trigger pagination events
       if (pagination && pagination.onChange) {
-        pagination.onChange(1, changeInfo.pagination!.pageSize);
+        pagination.onChange(1, changeInfo.pagination!.pageSize!);
```
Comment (afc163) [inline components/table/Table.tsx:221]: 为啥这里要补个 `!`。

Comment (JounQin) [reply components/table/Table.tsx:221]: `rc-pagination` 里的 pageSize 是必须的，而 `changeInfo.pagination` 里 `pageSize` 也有默认值

Comment (JounQin) [reply components/table/Table.tsx:221]: See https://github.com/ant-design/ant-design/runs/3575241779#step:5:24

**Pull request merged**